administrator/components/com_joomgallery/helpers/file.php

```diff
@@ -367,301 +383,409 @@ public static function resizeImage(&$debugoutput, $src_file, $dest_file, $usefor
-                            (int)$destHeight, $srcWidth, $srcHeight);
+          $meta_success = JoomFile::copyImageMetadata($src_file, $dest_file, $src_imagetype, $dest_imgtype);
+          if (!$meta_success)
+          {
+            $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_ERROR_COPY_METADATA');
+            return false;
+          }      
         }
-        if(!@imagejpeg($dst_img, $dest_file, $dest_qual))
+        if(!$success)
         {
           // Workaround for servers with wwwrun problem
           $dir = dirname($dest_file);
           JoomFile::chmod($dir, '0777', true);
-          imagejpeg($dst_img, $dest_file, $dest_qual);
+          $success = JoomFile::imageWriteFrom_GD($dest_file,$dst_img,$dest_qual,$dest_imgtype);
+          if ($metadata)
+          {
+            $meta_success = JoomFile::copyImageMetadata($src_file, $dest_file, $src_imagetype, $dest_imgtype);
+            if (!$meta_success)
+            {
+              $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_ERROR_COPY_METADATA');
+              return false;
+            }      
+          }
           JoomFile::chmod($dir, '0755', true);
         }
-        imagedestroy($src_img);
-        imagedestroy($dst_img);
+        foreach ($src_frames as $key => $frame)
+        {
+          imagedestroy($src_frames[$key]['image']);
+          imagedestroy($dst_frames[$key]);
+        }
         break;
       case 'gd2':
+        $debugoutput.='GD2...<br/>';
         if(!function_exists('imagecreatefromjpeg'))
+        // check, if GD is available
         {
           $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_LIBARY_NOT_INSTALLED');
           return false;
         }
         if(!function_exists('imagecreatetruecolor'))
+        // check, if GD2 is available
         {
           $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_NO_TRUECOLOR');
           return false;
         }
-        if($imginfo[2] == 'JPG')
+        // create empty image of specified size
+        $dst_frames = array();
+        if ($anim && $imginfo['animation'] && $imginfo['type'] == 'GIF')
         {
-          $src_img = imagecreatefromjpeg($src_file);
+          JLoader::register('GifFrameExtractor', JPATH_COMPONENT_ADMINISTRATOR . '/helpers/GifFrameExtractor.php');
+          $gfe = new GifFrameExtractor();
+          $src_frames = $gfe->extract($src_file);
+          foreach ($src_frames as $key => $frame)
+          {
+            $dst_frames[$key] = imagecreatetruecolor($destWidth, $destHeight);
+          }
         }
         else
         {
-          if($imginfo[2] == 'PNG')
-          {
-            $src_img = imagecreatefrompng($src_file);
-          }
-          else
-          {
-            $src_img = imagecreatefromgif($src_file);
-          }
+          $dst_frames[0] = imagecreatetruecolor($destWidth, $destHeight);
+          $src_frames = JoomFile::imageCreateFrom_GD($src_file, $dst_frames, $imginfo);
```

Comment:
Otherwise use the normal GD2 functions.

imageCreateFrom_GD: external function which serves as a replacement for imagecreatefromXXX. It creates a GD object from an existing image (from path) of a specific file-type